util/template/expression_template.go

```diff
@@ -151,3 +123,62 @@ func hasVarInEnv(env map[string]interface{}, parameter string) bool {
 	_, ok := flattenEnv[parameter]
 	return ok
 }
+
+// isTernaryExpression check if the expression is a ternary expression
+func isTernaryExpression(expression string) bool {
+	if strings.Contains(expression, "?") && strings.Contains(expression, ":") && len(expression) >= 5 {
```

Comment:
there's definitely some false positives here, for instance `workflow.parameters?.json ?? ['thing1', 'thing2'][1:]`

that's one reason why we should avoid parsing `expr` expressions on our own, as I mentioned above. a naive parse can cause a lot of issues